client.go

```diff
@@ -100,29 +97,24 @@ func (c *Client) Consumer(topics []string, cfg rdkafka.ConfigMap) (*Consumer, er
 	// topic
 	cid := ConsumerID(fmt.Sprintf("%s|%s", c.id, strings.Join(topics, ",")))
 
-	for _, topic := range topics {
-		if existingID, ok := c.consByTopic[topic]; ok {
-			if existingID != cid {
-				return nil, fmt.Errorf("Topic %s has another consumer", topic)
-			}
-		}
+	if c.consID == "" {
+		c.consID = cid
 	}
 
-	c.consumers[cid] = true
-	for _, topic := range topics {
-		c.consByTopic[topic] = cid
+	if cid != c.consID {
```

Comment:
I think the following version is easier to follow:

```golang
if cid != "" || cid != c.consID {
  return nil, fmt.Errorf("Client '%s' has already Consumer '%s' registered.", c.id, c.consID)
}

c.consID = cid

// ...
```

WDYT?

Reply:
This is an outdated commit; as mentioned in the respective commit message, `consID` was an intermediate step to allow a smoother drop of _ConsumerManager_.